python/pyspark/sql/types.py

```diff
@@ -384,6 +385,33 @@ def fromInternal(self, v: int) -> datetime.date:
             return datetime.date.fromordinal(v + self.EPOCH_ORDINAL)
 
 
+class TimeType(AtomicType):
```

Comment:
this PR actually add a new datatype in pyspark, not just python client.
It should work with both python client and pyspark classic.
We'd better also add test for pyspark classic.

Reply:
I'm not very professional with PySpark, this will take some time, I changed it to WIP by now.